src/util/server.ml

```diff
@@ -113,7 +113,8 @@ let analyze ?(reset=false) ({ file; do_analyze; _ }: t)=
     | Some solver_data ->
       let changes = CompareCIL.compareCilFiles file file in
       let old_data = Some { Analyses.cil_file = file; solver_data } in
-      { Analyses.changes; old_data; new_file = file }, false
+      (* TODO: get globals for restarting from config *)
```

Comment:
Shouldn't this logic be deduplicated?

Reply:
@sim642  You marked that as resolved --- I am not sure whether it is. Or is the idea that this should be deduplicated by #591?

Reply:
I think the TODO might be fine for the time being. The deduplication doesn't strictly need to happen in #591, but it's something we'll have to do soon anyway to a common reusable module for both. Ideally there would be no real logic in `Maingoblint` left, except for command-line arguments handling.